Content.Server/Singularity/EntitySystems/SingularitySystem.cs

```diff
@@ -21,56 +37,157 @@ public override void Initialize()
+                entity.HasComponent<ContainmentFieldComponent>() ||
+                entity.HasComponent<ContainmentFieldGeneratorComponent>()) return;
 
             // Singularity priority management / etc.
-            if (otherEntity.TryGetComponent<ServerSingularityComponent>(out var otherSingulo))
+            if (entity.TryGetComponent<ServerSingularityComponent>(out var otherSingulo))
+            {
+                // MERGE
+                if (!otherSingulo.BeingDeletedByAnotherSingularity)
+                {
+                    component.Energy += otherSingulo.Energy;
+                }
+
                 otherSingulo.BeingDeletedByAnotherSingularity = true;
+            }
 
-            otherEntity.QueueDelete();
+            entity.QueueDelete();
 
-            if (otherEntity.TryGetComponent<SinguloFoodComponent>(out var singuloFood))
+            if (entity.TryGetComponent<SinguloFoodComponent>(out var singuloFood))
                 component.Energy += singuloFood.Energy;
             else
                 component.Energy++;
         }
 
-        public override void Update(float frameTime)
+        /// <summary>
+        /// Handle deleting entities and increasing energy
+        /// </summary>
+        private void DestroyEntities(ServerSingularityComponent component, Vector2 worldPos)
         {
-            base.Update(frameTime);
-            _accumulator += frameTime;
+            // The reason we don't /just/ use collision is because we'll be deleting stuff that may not necessarily have physics (e.g. carpets).
+            var destroyRange = DestroyTileRange(component);
 
-            while (_accumulator > _updateInterval)
+            foreach (var entity in _lookup.GetEntitiesInRange(component.Owner.Transform.MapID, worldPos, destroyRange))
             {
-                _accumulator -= _updateInterval;
+                HandleDestroy(component, entity);
+            }
+        }
 
-                foreach (var singularity in ComponentManager.EntityQuery<ServerSingularityComponent>())
+        private void PullEntities(ServerSingularityComponent component, Vector2 worldPos)
+        {
+            // TODO: When we split up dynamic and static trees we might be able to make items always on the broadphase
+            // in which case we can just query dynamictree directly for brrt
+            var pullRange = PullRange(component);
+            var destroyRange = DestroyTileRange(component);
+
+            foreach (var entity in _lookup.GetEntitiesInRange(component.Owner.Transform.MapID, worldPos, pullRange))
+            {
+                // I tried having it so level 6 can de-anchor. BAD IDEA, MASSIVE LAG.
+                if (entity == component.Owner ||
+                    !entity.TryGetComponent<PhysicsComponent>(out var collidableComponent) ||
+                    collidableComponent.BodyType == BodyType.Static) continue;
+
+                if (entity.HasComponent<GhostComponent>() ||
```

Comment:
Even if we aren't adding SinguloEatableComponent yet you can probably condense this into a function for easy editing.

Reply:
I guess, even if it is only used in 1 spot.